src/services/mongoose/users.ts

```diff
@@ -88,6 +86,36 @@ export const UsersRepository = (): IUsersRepository => {
     }
   }
 
+  const migrateUserIdSubject = async ({
+    currentUserIdSubject,
+    newUserIdSubject,
+    phone,
+  }: {
+    currentUserIdSubject: UserId
+    newUserIdSubject: UserId
+    phone: PhoneNumber
+  }): Promise<User | RepositoryError> => {
+    const filter = { userId: { $eq: currentUserIdSubject } }
+    const update = {
+      $set: {
+        userId: newUserIdSubject,
+        phone,
+        deviceId: currentUserIdSubject,
+      },
+    }
+    const options = { upsert: true, returnOriginal: false }
+
+    try {
+      const result = await User.findOneAndUpdate(filter, update, options)
```

Comment:
yeah i tried that originally but it did not work because of this special use-case...updating the userId. calling something like this in the domain does not work:

```
UsersRepository().update({
    id,
    language,
    deviceTokens,
    phoneMetadata,
    phone,
    createdAt,
    deviceId,
  })
```

because downstream the service calls mongo like this:

```
const result = await User.findOneAndUpdate(
        { userId: id },
        {
          userId: id, // THIS DOES NOT UPDATE
          deviceTokens,
          phoneMetadata,
          language,
          phone,
          deviceId,
          createdAt, 
        },
        {
          new: true,
          upsert: true,
        },
      )
```

mongoose does not allow you to update the `userId` because we queried by it.

@nicolasburtey suggested we do it this way for now until we have the domain modeling session next time we touch the Auth code. just one of the trade-off with the architectural decision of making a  `userId` field either a `KratosId` or `DeviceId`